### DISCUSSION TITLE
Starting a thread resets window handle

### DISCUSSION THREAD

**Discussion by blayms** [dotnet/Silk.NET#1430] [Q&A] [answered]
I'm creating some [textures ](https://github.com/dotnet/Silk.NET/blob/1084fa66ce30f025622c5899a8059b595d4ff61c/examples/CSharp/OpenGL%20Tutorials/Tutorial%204.1%20-%20Model%20Loading/Texture.cs)on a thread.
When I start that "Texture creation" thread, window handle becomes 0 and get this exception on Texture ctor:

**Silk.NET.GLFW.GlfwException: 'NoContext: Cannot query entry point without a current OpenGL or OpenGL ES context'**

From what I know this is not the bug or anything like that. Most likely I need to implement multithreading in OpenGL, but idk how
Sorry if this question is stupid, but it's 04:45 AM, so I have no other choice

Any ways to fix it? I can't really find anyone experiencing this using Silk.NET

**Comment by Beyley** [comment, marked as answer]
You cannot run GL commands on a thread that does not have a GL context, and in the examples case, there is no context on your own thread

Theres not really a performant and easy way to do multithreading with OpenGL, so id recommend just doing it all on the main thread
